app/src/main/java/org/fossasia/openevent/general/MainActivity.kt

```diff
@@ -64,17 +62,11 @@ class MainActivity : AppCompatActivity() {
             R.id.favoriteFragment -> navAnimVisible(navigation, this@MainActivity)
             else -> navAnimGone(navigation, this@MainActivity)
         }
-        when (id) {
-            R.id.loginFragment,
-            R.id.signUpFragment -> navAnimVisible(navigationAuth, this@MainActivity)
-            else -> navAnimGone(navigationAuth, this@MainActivity)
-        }
     }
 
     override fun onBackPressed() {
         when (currentFragmentId) {
-            R.id.loginFragment,
-            R.id.signUpFragment -> {
+            R.id.authFragment -> {
                 navController.popBackStack(R.id.eventsFragment, false)
```

Comment:
This will always pop back to EventsFragment -> So if the user navigate from other fragments like Order or Ticket to log in and somehow they press back button, they will not return to the correct one.

Reply:
Yes, I have not removed it because imo it make scene. User is redirected to auth fragment where authentication requires, if user cancel the authentication then it should redirect to home screen.
If you want I will remove the condition from auth fragment and navigation flow will work in it's natural way?

Reply:
I also think that popping back to EventsFragment is fine in most case, except when a user is booking tickets in AttendeeFragment, canceling logging in will not return them to the current event/ticket detail that they're searching for.

Reply:
Okay so It will make complex to apply different conditions for different fragments. I will remove it. :+1:

Reply:
Any other changes you want to suggest?

Reply:
I don't. I think the rest is good

Reply:
I got a problem in that case. If the user goes to tickets, redirected to auth fragment. After pressing the back button, the user must redirect to events fragment.

Reply:
@iamareebjamal Can we use fragment's `navArgs()` in Main activity?

Reply:
Why would it be a problem?